gateway-messages/src/mgs_to_sp.rs

```diff
@@ -407,3 +409,23 @@ bitflags::bitflags! {
         const STARTUP_VERBOSE = 1 << 8;
     }
 }
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, SerializedSize,
+)]
+pub enum DumpRequest {
+    /// Returns the number of task crash dumps
+    TaskDumpCount,
+
+    /// Begins reading a task crash dump by index
+    ///
+    /// `key` is used for disambiguation between multiple readers
+    TaskDumpReadStart {
+        index: u32,
+        key: u32,
+    },
+
+    TaskDumpReadContinue {
+        key: u32,
```

Comment:
Does this need an offset to be idempotent under retries? If we had one, would we need a distinct `ReadStart` command vs sending a `ReadContinue` with offset 0? I guess the later would require resending the task index every time, which isn't ideal...

Reply:
Good catch, I added an explicit sequence number (which also requires support on the SP side)